Add an option to visualize Mastodon sphere size in Blender

maarzt opened the pull request:
The CSV exported by this plugin already contains the Mastodon "sphere radius".

This PR changes the `read_csv.py` and the geometry nodes in `empty_with_geometry_nodes.blend` to make use of the radius and correctly visualize it in Blender.

![image](https://github.com/mastodon-sc/mastodon-blender-view/assets/24407711/edf9de1b-9150-4842-8e0e-87ad91364e64)

Whether to show Mastodon sphere size or to show all spheres with the same size can be configured in Blender in the Modifier tab. (The settings can be seen in the screenshot above in the bottom ~~left~~ right corner. Click the small blue wrench icon.)

maarzt commented:
@stefanhahmann most changes here are on binary, so they are hard to review. But could you please do a functional test for this. That would be really nice.

stefanhahmann reviewed (approved):
I have tested with an example and I could see that the spheres in Blender are rendered with different sizes: 
![grafik](https://github.com/mastodon-sc/mastodon-blender-view/assets/10515534/51570485-1a72-45ee-812c-1551510359d7)
Thus, the intended functionality seems to work.

I was not able to set the option back to "all spheres with the same size". I do not see an option to set this in the "lower left corner" as mentioned in the PR. This is what Blender looks like on my machine directly after opening it:
![grafik](https://github.com/mastodon-sc/mastodon-blender-view/assets/10515534/6faf08e5-6622-4c58-9250-6564fbbfc26c)

* I do not know, where to make this setting in Blender. Some documentation might be good for this? On the other side: perhaps that documentation on how to do this is not really needed for actual users, since in many cases the behavior of the sphere with different radii will be the intended behavior by the users.

I have some more comments not really related to this PR particularly.

* I changed the blender version in the dialog `Set up Blender Addon...` (while having instable internet connection - perhaps the latter caused some trouble)
  * The dialog sometimes took really long to show up (sometimes more than a minute)
  * After changing from one Blender installation folder to another, it still showed `Addon is installed` and only changed it to `Addon is not installed` after a few seconds. This may be confusing to users.
  * The `Install addon` took quite long to install the addon (>1 minute) and did not provide any feedback about the installation process. I was not really sure, if something is happening
* The menu option to start the blender viewer is named `Open CSV in Blender`. This name is a bit confusing. I asked myself: which CSV and would I have to export it first? Maybe a better name would be: `Show tracks/spots in Blender`?
* I am wondering, why the blender view on my machine looks a bit edgy, whereas it the screenshot in the PR looks much smoother. Is there any settings that user needs to do achieve the smooth look?

maarzt commented:
> I was not able to set the option back to "all spheres with the same size". I do not see an option to set this in the "lower left corner" as mentioned in the PR.
>     * I do not know, where to make this setting in Blender. Some documentation might be good for this? On the other side: perhaps that documentation on how to do this is not really needed for actual users, since in many cases the behavior of the sphere with different radii will be the intended behavior by the users.

Sorry, it's the lower right corner. You need to: 1. select the "mesh" by clicking on one of the spheres. 2. In the lower right corner, click on the little blue wrench icon.
> 
> 
> I have some more comments not really related to this PR particularly:
> 
>     * I changed the blender version in the dialog `Set up Blender Addon...` (while having instable internet connection - perhaps the latter caused some trouble)
>       
>       * The dialog sometimes took really long to show up (sometimes more than a minute)
>       * After changing from one Blender installation folder to another, it still showed `Addon is installed` and only changed it to `Addon is not installed` after a few seconds. This may be confusing to users.
>       * The `Install addon` took quite long to install the addon (>1 minute) and did not provide any feedback about the installation process. I was not really sure, if something is happening

The only explanation I have is that Blender connects to some online resources, and this was taking really long!? I added a text in the setup dialog. Which informs the user that a internet connection is required for installing/updating the addon.

> 
>     * The menu option to start the blender viewer is named `Open CSV in Blender`. This name is a bit confusing. I asked myself: which CSV and would I have to export it first? Maybe a better name would be: `Show tracks/spots in Blender`?

I will introduce a sub menu for starting Blender. The menu will list the to possible options:
1. Interactive Blender Window
2. Geometry Nodes Blender Window

The difference between the two need to be explained in the online resources.

>     * I am wondering, why the blender view on my machine looks a bit edgy, whereas it the screenshot in the PR looks 
much smoother. Is there any settings that user needs to do achieve the smooth look?

Yes, it's actually the first setting in the settings panel, that you didn't find. The options is called "number of sphere subdivisions". It's a trade off between performance and smoothness of the spheres.

>     * Is there a way to start blender with the currently selected timepoint in mastodon?

TODO: open ticket

>       * I have started Blender with a project with 500+ timepoints. Played a bit around. Set timepoint to 300. Closed Blender. Closed Mastodon. Opened another project with just 30 timepoints. Opened Blender again. The timepoint was still set to 300 in Blender. It took me a while to understand the reason, why I did not see anything after these steps in Blender.

I'm surprised that Blender seems to remember that  timepoint. It will be another ticket: Start Blender with reasonable initial timepoint.

TODO: open ticket.

>     * I noticed that in the temp folder `read_csv.py` files accumulate. Would it be possible to delete them, when they are not needed anymore? Probably difficult...

Done. I used `File.deleteFileOnExit()` for those temp files. This should fix the problem. The temp files are read when start blender. So it's I guess ok to delete them even if Blender remains open.

>     * I would be convenient for developers who work with this Plugin, if there was a class similar to this under `src/main/test`, since the current available Start/Test scripts require some data to be existing in some specific paths:

I added a StartMastodonLauncher class.
